flask_weather_app.py

```diff
@@ -0,0 +1,63 @@
+from flask import Flask, render_template, request, send_file
+from config import openweather_api_key as api_key
+import requests
+import matplotlib.pyplot as plt
+from io import BytesIO
+
+app = Flask(__name__)
+
+def get_weather(city_name, forecast=False):
+    if not forecast:
+        base_url = "http://api.openweathermap.org/data/2.5/weather"
+    else:
+        base_url = "http://api.openweathermap.org/data/2.5/forecast"
+    params = {
+        'q': city_name,
+        'appid': api_key,
+        'units': 'metric'
+    }
+    response = requests.get(base_url, params=params)
+    if response.status_code != 200:
+        return None
+    return response.json()
+
+def plot_forecast(forecast):
+    if forecast:
+        times = [item['dt_txt'] for item in forecast['list']]
+        temps = [item['main']['temp'] for item in forecast['list']]
```

Comment:
the loop repeats here so you could make a single one, on the other hand it's probably small data so no impact and the list comprehensions are nice too